### PR TITLE
Addition of a singleton decorator class.

### PR DESCRIPTION
**Pull request by ejdecena**
This PR request incorporates the singleton decorator class.

### CODE REVIEW

**Comment by crazyguitar** [comment]
Cool !!!! 💯 I am impressed this snippet. Could you change title: singleton-decorator

I think singleton is a more general term. thx

File: docs/notes/python-object.rst [hunk 1]
```diff
@@ -363,6 +363,41 @@ Descriptor
     >>> hasattr(ex3, 'x')
     False
 
+Singleton
```
Comment (crazyguitar) [inline docs/notes/python-object.rst:366]: pls chang to "Singleton Decorator". thx

Comment (ejdecena) [reply docs/notes/python-object.rst]: Okay... Done :+1:

**Comment by crazyguitar** [comment]
Thx

**Pull request merged**